### PR TITLE
Update documentation for Chapter 20

### PR DESCRIPTION
**Pull request by gadmm**
No change entry needed.

### CODE REVIEW

File: manual/src/cmds/intf-c.etex [hunk 1]
```diff
@@ -2403,66 +2410,67 @@ section.
 \subsection{ss:c-thread-register}{Registering threads created from C}
 
 Callbacks from C to OCaml are possible only if the calling thread is
-known to the OCaml run-time system.  Threads created from OCaml (through
+known to the OCaml runtime system.  Threads created from OCaml (through
```
Comment (dra27) [inline manual/src/cmds/intf-c.etex:2413]: It would be _much_ easier to review PRs like this if stylistic changes like this could be in a separate PR, or at least in a separate commit. `git add -p` exists for a reason!

You've missed one on L847

File: manual/src/cmds/intf-c.etex [hunk 2]
```diff
@@ -1221,21 +1221,28 @@ has taken place since "r" was allocated.
...
 
 The variant \verb"caml_process_pending_actions_exn" is provided, that
-returns the exception instead of raising it directly into OCaml code.
-Its result must be tested using {\tt Is_exception_result}, and
-followed by {\tt Extract_exception} if appropriate. It is typically
-used for clean up before re-raising:
+does the same but returns the exception, instead of raising it
+directly into OCaml code. Its result must be tested using {\tt
+Is_exception_result}, and followed by {\tt Extract_exception} if
+appropriate. It is typically used for clean up before re-raising:
```
Comment (dra27) [inline manual/src/cmds/intf-c.etex:1245]: I think this semantically neutral - personally, I'm not sure it's necessary (the function is referred to as a "variant" so it doesn't really need the addition of "does the same but" to me)

File: manual/src/cmds/intf-c.etex [hunk 3]
```diff
@@ -2403,66 +2410,67 @@ section.
...
+concurrently with the C code of the primitive on this domain.
 
 If a C primitive runs for a long time or performs potentially blocking
-input-output operations, it can explicitly release the master lock,
-enabling other OCaml threads to run concurrently with its operations.
-The C code must re-acquire the master lock before returning to OCaml.
-This is achieved with the following functions, declared in
-the include file "<caml/threads.h>".
+input-output operations, it can explicitly release the domain lock,
+enabling other threads to run concurrently with its operations. The C
```
Comment (dra27) [inline manual/src/cmds/intf-c.etex:2448]: I think this needs clarifying - releasing the domain lock allows other systhreads attached to the domain to execute OCaml code, but systhreads attached to other domains are already able to execute in parallel. It's probably a good opportunity to start using "parallel" vs "concurrent" (domains/effects confusion, etc.)

File: manual/src/cmds/intf-c.etex [hunk 4]
```diff
@@ -2522,16 +2529,16 @@ after:
 \end{verbatim}
 
 Note: the "acquire" and "release" functions described above were
-introduced in OCaml 3.12.  Older code uses the following historical
-names, declared in "<caml/signals.h>":
+introduced in OCaml 3.12. The following equivalent historical names
+are declared in "<caml/signals.h>" and are sometimes used instead:
```
Comment (dra27) [inline manual/src/cmds/intf-c.etex:2533]: I don't think this change is necessary - apart from anything else, it suggests that `caml_enter_blocking_section` is a suitable alias as opposed to an older one (although I confess I still use it...)

**Review by dra27 — COMMENTED** [review]
Thanks for starting this - I think it's probably worth going further with the distinction on systhreads and how that interacts with the domain lock (further comments in the PR).

Separately, can I please beseech you _either_ to resist the temptation to edit existing text (line breaks, double-spaces after full-stops, s/run-time/runtime, etc.) _or_ use `git add -p` to put them in separate commits. As a reviewer, I'd really like to look at a diff where everything in red and green is critical to review, not subject to artistic interpretation of English punctuation and grammar!

File: manual/src/cmds/intf-c.etex (hunk 1 above)
Comment (gadmm) [reply manual/src/cmds/intf-c.etex:2413]: I am familiar with the `--patch` command line and I frequently use it to split my commits in units of sense when appropriate. I think the present commit was simple enough.

> You've missed one on L847

In _"computed at run-time"_, we have an adjective formed from two words separated with a dash. This is proper English, unlike _"computed at runtime"_.

Note that while the change is stylistic (_"run-time system"_ is perfectly correct English), most of the documentation already uses "runtime system" instead.

File: manual/src/cmds/intf-c.etex (hunk 2 above)
Comment (gadmm) [reply manual/src/cmds/intf-c.etex:1245]: This is stylistic, but having written the original sentence I can decide to change it.

File: manual/src/cmds/intf-c.etex (hunk 3 above)
Comment (gadmm) [reply manual/src/cmds/intf-c.etex:2448]: Well yes, that's why it says "concurrently".

File: manual/src/cmds/intf-c.etex (hunk 4 above)
Comment (gadmm) [reply manual/src/cmds/intf-c.etex:2533]: The previous text is outdated. `caml_enter/leave_blocking_section` are more used than the alternative and are now better-documented. They have the `no_pending` variant, which was not added with a "release" alias. So it needs to be changed so that it does not make it sound like most of the code on opam uses a bad/obsolete alias.

**Comment by gadmm** [comment]
I have spent more time on this PR than allocated, and I am not seeing it converging, so I have to close the PR. I am leaving my branch online so that someone can pick it up.
